helix-view/src/graphics.rs

```diff
@@ -92,6 +92,57 @@ impl Rect {
         self.y.saturating_add(self.height)
     }
 
+    // Returns a new Rect with width reduced from the left side.
+    // This changes the `x` coordinate and clamps it to the right
+    // edge of the original Rect.
+    pub fn chop_from_left(self, width: u16) -> Rect {
+        let width = std::cmp::min(width, self.width);
+        Rect {
+            x: self.x.saturating_add(width),
+            width: self.width.saturating_sub(width),
+            ..self
+        }
+    }
+
+    // Returns a new Rect with width reduced from the right side.
+    // This does _not_ change the `x` coordinate.
+    pub fn chop_from_right(self, width: u16) -> Rect {
+        Rect {
+            width: self.width.saturating_sub(width),
+            ..self
+        }
+    }
+
+    // Returns a new Rect with height reduced from the top.
+    // This changes the `y` coordinate and clamps it to the bottom
+    // edge of the original Rect.
+    pub fn chop_from_top(self, height: u16) -> Rect {
+        let height = std::cmp::min(height, self.height);
+        Rect {
+            y: self.y.saturating_add(height),
+            height: self.height.saturating_sub(height),
+            ..self
+        }
+    }
+
+    // Returns a new Rect with height reduced from the bottom.
+    // This does _not_ change the `y` coordinate.
+    pub fn chop_from_bottom(self, height: u16) -> Rect {
+        Rect {
+            height: self.height.saturating_sub(height),
+            ..self
+        }
+    }
+
+    pub fn with_height(self, height: u16) -> Rect {
+        // new height may make area > u16::max_value, so use new()
+        Self::new(self.x, self.y, self.width, height)
+    }
+
+    pub fn with_width(self, width: u16) -> Rect {
+        Self::new(self.x, self.y, width, self.height)
+    }
+
     pub fn inner(self, margin: &Margin) -> Rect {

```

Comment:
I was also thinking of renaming `inner` to `with_margin`

Reply:
`with_margin` seems to imply to me that it's the (inner) area + margin

Reply:
Ah true, I'll leave it as is.